### PR TITLE
 #175743408 ft(notification):  implement and integrate notifications

### PR DESCRIPTION
**Pull request by SimonImanigirimpuhwe**
#### What does this PR do?

- Integrate notifications 

#### Description of Task to be completed? 

- Design and implement notification pane
- Set notification context
- Install socket.io-client
- Connect frontend to the backend

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch **`ft-implement-notifications-pane-#175743408`**

- Then run **`npm install`** to install all dependencies

- Run **`npm start`** to start the server

- In your default browser navigate to **`http://localhost:3000/signin`** to login 

> When successfully reached the dashboard you should see notification icon and click on it to view current notifications

#### Any background context you want to provide?

- To successfully send request to the backend you need to put backend url in `.env` file as **`REACT_APP_BACKEND_URL`**

#### What are the relevant Pivotal Tracker stories?

- [#175743408](https://www.pivotaltracker.com/n/projects/2476677/stories/175743408)
- [#175743417](https://www.pivotaltracker.com/n/projects/2476677/stories/175743417)
- [#175743421](https://www.pivotaltracker.com/n/projects/2476677/stories/175743421)

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A

### CODE REVIEW

File: src/views/Login/Login.js [hunk 1]
```diff
@@ -45,38 +46,50 @@ export default function SignIn() {
...
+           });
+           const jsonData = await response.json();
+
+           if(jsonData.user !== undefined){
+               localStorage.setItem("userId",jsonData.user.id);
+               localStorage.setItem("userToken",jsonData.token);
+               localStorage.setItem("userImageUrl",jsonData.user.user_image);
+               dispatch({type: SET_LOADING, payload: false })      
+               dispatch({type: SET_AUTHENTICATION, user:jsonData.user, token:jsonData.token })
+               toaster(jsonData.message, 'success')
```
Comment (SamuelSabasNayo) [inline src/views/Login/Login.js]: This toaster is not working during successful login.

File: src/views/Login/Login.js [hunk 2]
```diff
@@ -131,7 +144,6 @@ export default function SignIn() {
             labelWidth={70}/>
         </InputText>
 
-
           <FormControlLabel
```
Comment (SamuelSabasNayo) [inline src/views/Login/Login.js]: It does not remember a logged-in user.

**Review by SamuelSabasNayo — CHANGES_REQUESTED** [review]

File: src/views/Login/Login.js (hunk 1 above)
Comment (SimonImanigirimpuhwe) [reply src/views/Login/Login.js]: Let me see what might be the cause.

File: src/views/Login/Login.js (hunk 2 above)
Comment (SimonImanigirimpuhwe) [reply src/views/Login/Login.js]: I am not the one who did login integration but when I was doing my work it was working as it should.
Let me inspect to see what might be the cause.

**Review by SamuelSabasNayo — APPROVED** [review]
Please rebase to develop.

**Review by furebo — APPROVED** [review]

**Review by furebo — COMMENTED** [review]
But there are some conflicts to resolve before it can be marged

**Review by kabano89 — APPROVED** [review]

**Review by leny62 — APPROVED** [review]

**Pull request merged**